lib/explorer/polars_backend/shared.ex

```diff
@@ -42,36 +42,55 @@ defmodule Explorer.PolarsBackend.Shared do
   def apply_dataframe(%DataFrame{} = df, %DataFrame{} = out_df, fun, args) do
     case apply(Native, fun, [df.data | args]) do
       {:ok, %module{} = new_df} when module in @polars_df ->
-        if @check_frames do
-          # We need to collect here, because the lazy frame may not have
-          # the full picture of the result yet.
-          check_df =
-            if match?(%PolarsLazyFrame{}, new_df) do
-              {:ok, new_df} = Native.lf_collect(new_df)
-              create_dataframe(new_df)
-            else
-              create_dataframe(new_df)
+        {struct?, dtypes} =
+          if @check_frames do
+            # We need to collect here, because the lazy frame may not have
+            # the full picture of the result yet.
+            check_df =
+              if match?(%PolarsLazyFrame{}, new_df) do
+                {:ok, new_df} = Native.lf_collect(new_df)
+                create_dataframe(new_df)
+              else
+                create_dataframe(new_df)
+              end
+
```

Comment:
DF.mutate for struct related ops are special case as we may not know dtype ahead of op. I am just accepting the returned dtype. I made sure that the exception is narrowed down to  :df_mutate_with_exprs and only happens for structs.

Reply:
Hrm, this can be a big deal and a blocker for us. :( We need to know the operations so we can build lazy queries, which make sense to be impossible in this case. So we'd need to require the dtypes.

Other options are:

1. Introduce `DF.from_json` and add a mechanism to convert a DF to a series (i.e. a struct) (either DF.to_structs or Series.from_dataframe) (this means we can support infer schema length and dtype options as the other conversion functions)

2. Add `Series.from_json`

I am more inclined to go with option 1 for now. Thoughts?

Reply:
I think supporting option 2 is a low hanging fruit: 

- raise error when from_json is used as data frame operation
- rename `json_decode`  to `from_json`

```elixir
d = DF.new([%{a: "{\"n\": 1}"}])
aj = Series.from_json(df["a"])
DF.put(df, "aj", aj) 
```